pennylane/tape/qscript.py

```diff
@@ -461,7 +461,8 @@ def _update_par_info(self):
         for idx, m in enumerate(self.measurements):
             if m.obs is not None:
                 self._par_info.extend(
-                    {"op": m.obs, "op_idx": idx, "p_idx": i} for i, d in enumerate(m.obs.data)
+                    {"op": m.obs, "op_idx": idx + len(self.operations), "p_idx": i}
```

Comment:
I made this change to better keep track of which entries in `_par_info` corresponds to which operators/measurements via `op_idx`. (Previously it could be the case that the `op_idx` overlaps between operators and measurements.)

Also curious about what consequences this has, although `_par_info` doesn't seem to be used outside this file.

Reply:
it's currently used in the hadamard gradient and pulse gradient (retrieved via `tape.get_operation(some_trainable_param)`, but those gradients implicitly assume that the measurements are _not_ trainable. We could probably make them bug out if we give it a trainable observable, and maybe we should to test this bug! I think this is a sign that `_par_info` needs some love

Reply:
I was trying out what would happen to these gradient transforms if the observables were trainable, and I got a ton of cryptic error messages both on `master` and on this branch. So I'd rather save myself the headache and revert this, at the cost of making `QuantumScript.set_parameters` a bit more complicated.